Foxtrot/Driver/Rewriting/Utils/TypeNodeExtensions.cs

```diff
@@ -0,0 +1,20 @@
+﻿using System.Compiler;
+using System.Diagnostics.Contracts;
+
+namespace Microsoft.Contracts.Foxtrot.Utils
+{
+    internal static class TypeNodeExtensions
+    {
+        public static int TemplateArgumentsCount(this TypeNode typeNode)
```

Comment:
:thought_balloon: I don't like this extension method for the same reason as `CountOrDefault`. The way to write this is the following:

```
typeNode.TemplateArguments?.Count ?? 0;
```